### ISSUE TITLE
Cannot handle classification labels other than sequential values from 0 to N

### ISSUE THREAD

**Issue by nickwp** [WatChMaL/WatChMaL#52]
PyTorch's cross entropy loss function requires that the class labels are from 0 to C-1 where C is the number of classes.
We need to be able to handle arbitrary class labels.

**Comment by nickwp** [comment]
It should be relatively straightforward to internally map an arbitrary set of labels to the range 0 to C-1, however we will need to decide how to handle output. Currently the softmax output is an array where one dimension corresponds to the class label, so if the labels are no longer from 0 to C-1 then we should clearly define how we arrange this dimension of the softmax outputs, or provide an alternative way of outputting the softmax values.

**Comment by nickwp** [comment]
This is now fixed by #58
